### PR TITLE
Examples code re-written with response.body stream converted to string.

### PR DESCRIPTION
**Pull request by igorbalden**
#### Summary
Edited code examples, to convert sequence of two .then() functions, to a single one.

#### Motivation
According to the examples the developer can access all 'response' properties in the first .then(), 
and the response.body data in the second .then(), separately.
With the suggested code the whole 'response' object is available in the first .then() block.
That is all response properties, together with the response.body.

#### Supporting details
Lots of developers do not use fetch API because of this exact issue.
They can not access together response properties and data in one code block.
The examples code confuses users, as it is obvious by the stackoverflow.com questions.

#### Metadata

- [ ] Adds a new document
- [x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by domenic** [comment]
This seems strange to me. Either use async/await entirely (no `.then()`), or avoid `async`/`await` entirely (like the existing code does).

**Comment by igorbalden** [comment]
There is no problem mixing .then(), with async/await syntax. It's only a matter of style.
Anyway, here they are examples in both 'single' styles:
1.  .then()
    We have to return the Promise in the first .then(),
    else we won't be able to catch all errors in a single catch();

  function fetchit() {
    fetch('https://some/resource')
    .then(response => {
      return Promise.resolve(response.json())
      .then(data => {
        console.log(response)
        console.log(data);
      })
    })
    .catch(error => {
      console.log(error)
    });
  }

2.  async/await 
    Again we may catch all errors in a single catch();

  async function fetchit() {
    try {
      const response = await fetch('https://some/resource');
      const data = await response.json();
      console.log(response)
      console.log(data)
    } catch(error) {
      console.log(error)
    };
  }

I still believe that the mixed style version is the most convenient of all, and must be included.

Would you like me to rewrite the "Using fetch" page, to include all these examples?

**Comment by sideshowbarker** [comment]
Thanks for opening this PR. I agree this article has a lot of room for improvement, so I’m glad to see it getting some attention.

>  The examples code confuses users, as it is obvious by the stackoverflow.com questions.

Yes, I read many questions posted to the Stack Overflow `[fetch-api]` tag, and I agree there’s a lot of developer confusion about how to write basic code using the Fetch API.

> Either use async/await entirely (no `.then()`), or avoid `async`/`await` entirely (like the existing code does).

I tend to agree with Domenic on this. In terms of developer confusion, I think showing developers a mix of `.then()` and `async`/`await` in a basic code example is potentially confusing for developers. I can imagine some newer developers thinking it means that *must* write `fetch()` code in that mixed style for some reason.

> There is no problem mixing .then(), with async/await syntax.

I don’t think anybody was suggesting there’s a problem with how it executes. But it seems fair to say it’s not typical.

> It's only a matter of style.

True — but our existing style in MDN is generally not to mix them. Personally, I see some value in the mixed approach, and I understand why you’d prefer it, and I can imagine there are other experienced developers who would prefer it too — developers who know what they’re doing and choose to do it that way. But with an MDN article like this, we’re kind of optimizing for the developers who may not yet know well what they’re doing — and so we somewhat need to try to keep thing as simple a possible, and not introduce anything that might further confuse anybody new.

> Would you like me to rewrite the "Using fetch" page, to include all these examples?

IMHO, it’s probably sufficient to just show one example of the async/await approach after the “_A basic fetch request is really simple to set up. Have a look at the following code:”_ line, to follow the existing `.then()`-based approach there.

**Comment by igorbalden** [comment]
Tnx @domenic , @sideshowbarker for your guidance.
Pushed on more commit. 
Edited all examples in .then() syntax, 
and added one example of sync/await.

File: files/en-us/web/api/fetch_api/using_fetch/index.md [hunk 1]
```diff
@@ -63,12 +77,17 @@ async function postData(url = '', data = {}) {
     referrerPolicy: 'no-referrer', // no-referrer, *no-referrer-when-downgrade, origin, origin-when-cross-origin, same-origin, strict-origin, strict-origin-when-cross-origin, unsafe-url
     body: JSON.stringify(data) // body data type must match "Content-Type" header
   });
-  return response.json(); // parses JSON response into native JavaScript objects
+  return response;
```
Comment (jpmedley) [inline files/en-us/web/api/fetch_api/using_fetch/index.md:80]: What's your reason for remving `json()` from this example, while adding it to the next example.

File: files/en-us/web/api/fetch_api/using_fetch/index.md [hunk 2]
```diff
@@ -231,19 +256,17 @@ run();
 ```
 
 ### Checking that the fetch was successful
-
-A {{domxref("fetch()")}} promise will reject with a {{jsxref("TypeError")}} when a network error is encountered or CORS is misconfigured on the server-side, although this usually means permission issues or similar — a 404 does not constitute a network error, for example. An accurate check for a successful `fetch()` would include checking that the promise resolved, then checking that the {{domxref("Response.ok")}} property has a value of true. The code would look something like this:
-
+A {{domxref("fetch()")}} promise will reject with a {{jsxref("TypeError")}} when a network error is encountered or CORS is misconfigured on the server-side, although this usually means permission issues or similar — a 404 does not constitute a network error, for example. An accurate check for a successful `fetch()` would include checking that the promise resolved, then checking that the {{domxref("Response.ok")}} property has a value of true. The code would look something like this:
```
Comment (jpmedley) [inline files/en-us/web/api/fetch_api/using_fetch/index.md:259]:
```suggestion
A {{domxref("fetch()")}} promise will reject with a {{jsxref("TypeError")}} when a network error is encountered or CORS is misconfigured on the server side, although this usually means permission issues or similar. A 404 does not constitute a network error, for example. An accurate check for a successful `fetch()` would include checking that the promise resolved, then checking that the {{domxref("Response.ok")}} property has a value of true. The code would look something like this:  
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

**Comment by wbamberg** [comment]
I'm closing this PR. The changes make the examples harder to understand and it's not obvious to me why we should in general prefer the style shown here.

**Pull request closed without merging**